ops/model.py

```diff
@@ -1418,17 +1418,17 @@ def can_connect(self) -> bool:
             #  instance that is in fact 'ready'.
             self._pebble.get_system_info()
         except pebble.ConnectionError as e:
-            logger.debug("Pebble API is not ready; ConnectionError: %s", e.message())
```

Comment:
so this feels like it would end up with redundant information, as e.message() was just the error string, while str(e) is getting mapped to e.__repr__() and that includes things like the Type again so you would end up with:
`Pebble API is not ready; ConnectionError: <ops.pebble.ConnectionError, "This is my connection error">`

Reply:
No, `%s` will map to `str(e)`, but that's not the same as `repr(e)` -- I guess because the base `Exception` type has a `__str__` which string-formats its arguments, or just `str(arg)` if there's only one.

```
$ cat s.py
import urllib.request

from ops import pebble

try:
	try:
		urllib.request.urlopen('foo://bar.com')
	except urllib.error.URLError as e:
		raise pebble.ConnectionError(e.reason)
except pebble.ConnectionError as e:
	print("Pebble API is not ready; ConnectionError: %s" % e)
$ python3 s.py
Pebble API is not ready; ConnectionError: unknown url type: foo
```

But if you explicitly use repr (change the `%s` to a `%r` above), you get what you had feared:

```
$ python3 r.py 
Pebble API is not ready; ConnectionError: <ops.pebble.ConnectionError ('unknown url type: foo',)>
```